### PR TITLE
gh-102541: Hide traceback in help prompt

### PR DESCRIPTION
**Pull request by Eclips4**
Add `is_cli` param to `doc` function (by default, set to `False`).
When `is_cli` arg is `True`, we raise a ImportError, otherwise, just show a `missing_pattern`.
Although, add a variable `missing_pattern`, which now is used by `resolve` and `doc` functions.
This PR absolutely compatible with #89130.
I like tests, but i think, that case shouldn't be tested.
If core-devs approve this, it should be merged to `3.11` and `3.12` branches. 
On 3.10 this works fine.

* Issue: gh-102541

### CODE REVIEW

**Comment by Eclips4** [comment]
Could someone review it? =)

File: Lib/pydoc.py
Comment (chgnrdv) [inline Lib/pydoc.py]: You can re-raise `ImportError` that is raised in `resolve` called in `render_doc` instead of constructing the same exception again. Also, instead of invoking `pager` screen only to show message about missing `thing`, you can print this message directly to stdout, as 3.10 does.
```python
try:
    pager(render_doc(thing, title, forceload))
except ImportError as e:
    if is_cli:
        raise
    print(e)
```

Comment (chgnrdv) [reply Lib/pydoc.py]: (I am not a core dev to review though, just trying to help. Maybe you should ping someone who is responsible for this code)

File: Lib/pydoc.py
Comment (chgnrdv) [inline Lib/pydoc.py:1787]: Instead of invoking `pager` screen just to show message about missing `thing`, you can print this message directly to stdout, just as 3.10 does.
```python
print(missing_pattern % thing)
```

File: Lib/pydoc.py
Comment (Eclips4) [reply Lib/pydoc.py]: Thanks, @chgnrdv !
Yeah, that's look right, i'll soon change this.

Just FYI, when you reviewing someone `PR`, on the page `Files changed` you can click a button called `Add a suggestion`. This is a more convenient way in this case. =)

Comment (chgnrdv) [reply Lib/pydoc.py]: Thank you, this is really helpful. I saw this thing being done by reviewers and wondered how to do it myself :-)

Your code looks good to me, but you have to wait for someone who is authorized for review. This might take some time since the number of contributors far exceeds the number of reviewers.

**Comment by Eclips4** [comment]
@sobolevn 
Can you review it?

**Review by chgnrdv — APPROVED** [review]

**Comment by Eclips4** [comment]
Can someone from core developers review it?

**Comment by terryjreedy** [comment]
The most appropriate coredev reviewer would be whoever merged the PR that caused the regression.  That would have to be a feature addition after 3.10.0b1, May 2021 (and hence not backported to 3.10) and before 3.11.0 in Oct 2022.  #98450 fixed a related bug --  `help('invalid query')` raising with similar result.  This caught the exception in a different place (line 2005) and remains fixed.

**Comment by corona10** [comment]
I will take a look at this PR tonight (UTC+9)

File: Lib/pydoc.py
Comment (corona10) [inline Lib/pydoc.py]: Please revert the unnecessary changes.

**Review by corona10 — APPROVED** [review]
LGTM with nit comment.
Writing test code will be great, but I know this case is very hard to write. :)

**Comment by corona10** [comment]
Please rebase the PR.

File: Lib/pydoc.py
Comment (Eclips4) [reply Lib/pydoc.py]: No problems, but I have one question:
Which changes you consider as unnecessary?
Just this indent? Indeed, I did it by accident. 
What do you think about new variable `missing_pattern`? Should I revert this change also?

Comment (corona10) [reply Lib/pydoc.py]: > What do you think about new variable missing_pattern? Should I revert this change also?

Yes please, if possible, I prefer the minimal change as possible to track the previous history when we need it. 

> Which changes you consider as unnecessary?

Most of cases, unrelated space changes... something like that.

File: Misc/NEWS.d/next/Library/2023-03-12-01-17-15.gh-issue-102541.LK1adc.rst
Comment (corona10) [inline Misc/NEWS.d/next/Library/2023-03-12-01-17-15.gh-issue-102541.LK1adc.rst]:
```suggestion
Hide traceback in :func:`help` prompt, when import failed.
```

**Pull request merged**
